### DISCUSSION TITLE
Did memory managment change with new updates?

### DISCUSSION THREAD

**Discussion by usigg3** [vladmandic/sdnext#386] [Q&A]
Have a RTX 2060 with 6 GB and a day ago I could generate with 2 controlnets enabled softedge, and depth. Now the exact same generation throws out of memory errors and crashes. Is there some way to optimize memory like the old --medvram?

**Comment by vladmandic** [comment]
behavior of `--lowvram` or `--medvram` has not changed, but everything about CUDA else did: check UI -> Settings -> CUDA Params

**Reply by usigg3** [reply]
So --medvram still works. I thought I read somewhere that it was ignored?

**Reply by vladmandic** [reply]
nope, its still there. things like `--no-half` were moved recently.
eventually, `--medvram` will also move, but not planned just yet.
my goal is that everything should be tunable from ui and usage of cmdflags should be only for troubleshooting and exceptions.

**Reply by usigg3** [reply]
maybe there is something with depth controlnet then... This is what I get with only depthleres. It worked yesterday...

gradio call: RuntimeError
╭───────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────╮
│ E:\vwebui\automatic\modules\call_queue.py:58 in f                                                            │
│                                                                                                              │
│    57 │   │   │   │   pr.enable()                                                                            │
│ ❱  58 │   │   │   res = list(func(*args, **kwargs))                                                          │
│    59 │   │   │   if shared.cmd_opts.profile:                                                                │
│                                                                                                              │
│ E:\vwebui\automatic\modules\call_queue.py:38 in f                                                            │
│                                                                                                              │
│    37 │   │   │   try:                                                                                       │
│ ❱  38 │   │   │   │   res = func(*args, **kwargs)                                                            │
│    39 │   │   │   finally:                                                                                   │
│                                                                                                              │
│                                           ... 19 frames hidden ...                                           │
│                                                                                                              │
│ E:\vwebui\automatic\venv\lib\site-packages\torch\nn\modules\module.py:820 in _apply                          │
│                                                                                                              │
│    819 │   │   │   with torch.no_grad():                                                                     │
│ ❱  820 │   │   │   │   param_applied = fn(param)                                                             │
│    821 │   │   │   should_use_set_data = compute_should_use_set_data(param, param_applied)                   │
│                                                                                                              │
│ E:\vwebui\automatic\venv\lib\site-packages\torch\nn\modules\module.py:1143 in convert                        │
│                                                                                                              │
│   1142 │   │   │   │   │   │   │   non_blocking, memory_format=convert_to_format)                            │
│ ❱ 1143 │   │   │   return t.to(device, dtype if t.is_floating_point() or t.is_complex() else No              │
│   1144                                                                                                       │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
RuntimeError: [enforce fail at ..\c10\core\impl\alloc_cpu.cpp:72] data. DefaultCPUAllocator: not enough memory:
you tried to allocate 58982400 bytes.

**Reply by utak3r** [reply]
I'm also on 6GB card and also getting lots of errors. Initially I thought this is due to I've updated my torch to 2.0, so reinstalled 1.13.1 - but it's still like that. Will be investigating the settings then. Will let you know.

**Reply by usigg3** [reply]
After a complete restart it works for a few hundred generations then it will crash trying to allocate more memory. Looks like some sort of memory leak...

**Reply by utak3r** [reply]
> I'm also on 6GB card and also getting lots of errors. Initially I thought this is due to I've updated my torch to 2.0, so reinstalled 1.13.1 - but it's still like that. Will be investigating the settings then. Will let you know.

I've setup a new venv from scratch - and it's much better for now. Will continue testing...

**Comment by Threnos** [comment]
@vladmandic 
I have a 4Gb vram and it heavily impacts me. With --medvram I used to be able to generate batchSize:8 512x512 images using UniPC and don't hit the ceiling with a head, now only 3. Upon startup, python occupies 880 Mb of vram, when it used to be around 150.

**Comment by Threnos** [comment]
This day I was generating non-stop using xyz script (set sampllers and checkpoint names) and "generate forever". It have been working smoothly for ~10min, then I went afk for ~15-20min. And when I returned to check the process, it was only throwing "out of vram" errors. It looks like a memory leak (@usigg3 already mentioned it)

**Reply by razvan-nicolae** [reply]
This explains the issues ive seen with 4090 large resolution generations too. I swear I’ve had up to 10gb reserved for 1x 512px generations at one point. And after a restart it would be back to normal.

My guess is they happen quite fast in my case, i never needed hundreds to reach this point. , It’s not a complete leak but its limited by the allocation in pytorch, usually 15-17gb of vram as I’ve seen. Then you can still work with around 6-8, but it’s pretty sad.
Tested side by side with old a1111 and it never happens there so its not a torch 2.0 issue

**Reply by Threnos** [reply]
Yeah, my 3050ti 4Gb needs quite more time to process)

**Comment by mykeehu** [comment]
There must be a memory leak with the Hires fix or/and ControlNet. For the second day now, I've noticed that when I do Foolhardy Remacri 2x upscale under txt2img with ControlNet, I'm running out more and more VRAM. Eventually it fills up and uses normal RAM. 
I notice it because it gets slower to generate. In this case I disconnect one of the 3 ControlNet layers and gain some VRAM, but instead of the initial 4-6 GB it not infrequently uses almost 13. I haven't been able to track down if it's the ControlNet or the Hires fix that's the problem, I only notice it when it slows down (after about 1-2 hours).

**Reply by Threnos** [reply]
I don't use hiresFix (even hidden it from the ui) and have ControlNet extention disabled, but still experiencing vram leak

**Reply by vladmandic** [reply]
how big is the vram leak? does it grow with each batch generated?

**Reply by mykeehu** [reply]
If not with every generation, I notice that it gets slower. I look at the VRAM and see it's already over 12 GB, so it's using RAM. I reboot the SD system and it's good again, around 6 GB. I'll watch it more closely in the days to come, because it seems to be increasing bit by bit, so it's hard to figure out what's causing it.

**Reply by Threnos** [reply]
It's not easy to tell right away.
Hold on... I just got a pale clue, give me time to confirm or bust my theory.
The clue is that with xyz checkpoint switching might prevent the leak.
Finally my professional skills are put to community benefit!

**Comment by Threnos** [comment]
What I can tell for sure, is that launching with params: sampler UniPC (bh2, logSNR, lower order final, order 3), steps 44, 1024x1024, batch count and size 1, cfg 7, seed -1, no scripts or extentions. Instantly fills up all memory. Using --medvram.

**Comment by mykeehu** [comment]
OK, Controlnet is probably to blame, there is already a discussion about this:
https://github.com/Mikubill/sd-webui-controlnet/issues/990

**Reply by Threnos** [reply]
Hmmm, is it possible that if an extention is disabled, that it still can execute code?

**Reply by anothertal3** [reply]
Good find. I also noticed memory issues since very recently but haven't been using controlnet at all. I will attempt to disable the extension as a workaround.

**Reply by utak3r** [reply]
Hm, I'm using ControlNet in A1111 all the time without issues...

**Reply by qpl23** [reply]
That controlnet discussion seems to highlight a system RAM (mainboard memory) rather than VRAM (gpu memory) issue?

Regards that, I've seen persistent jumps in RAM use on my (linux/AMD) setup, especially but not exclusively when switching checkpoints, to the point of locking the machine or turning performance to molasses, looking very like a memory leak somewhere.

After applying a [suggestion from an automatic1111 issue](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/9323#issuecomment-1499832668) this problem pretty much went away, so I recommend looking into libtcmalloc and LD_PRELOAD as per the linked comment for anyone suffering RAM issues on linux (check with ```free -h``` command after generations or checkpoint swaps.)

**Reply by mykeehu** [reply]
I'm on Windows, and I usually use these:
- TI 
- Lora 
- Controlnet 
- Hires fix 

Usually they are memory consuming separately, but for VRAM, 12 GB used to be enough. Now by the time I run Hires fix, it's already taking up regular RAM, because there's no space left in VRAM.

At the end of the operations, only Hires fix is taken out of VRAM, the rest is held, not freed by the system. 
For example, if I look at it without Hires fix: starts from 8 GB, goes up to 9.5, then at the end of the operation stays at 8.6.
I then turned off Additional Networks (so no Lora). Generate again from 8.5 to 9.5, then back to 8.5 again at the end.
So with Lora, I gained nothing. Would it have remained in the video memory?
I turned off Controlnet, but the memory is the same as before. I have no more ideas.

**Reply by anothertal3** [reply]
Lucky accident or not: I haven't experienced strange behaviors after I've recently disabled the controlnet extension.

**Reply by vladmandic** [reply]
@anothertal3 would be great if we can confirm that controlnet is cause of memory leak.

**Reply by anothertal3** [reply]
Understood. Will try to reproduce the issue after reenabling the extension and report back.

**Reply by anothertal3** [reply]
It's not really a proof, but:
- When controlnet was disabled I was able to process img2img from 512x768 to 1024x1536 at any time
- After enabled controlnet it's pretty much impossible due to the fact that "system info" tells me the following `gpu: free:0.28 used:5.72 total:6.0` which would always result in a "OutOfMemoryError"

**Reply by vladmandic** [reply]
And new CN models are even more hungry, but that's not the issue, question is if it's leaking. That's where new built-in monitor should she'd some light.

**Comment by lolxdmainkaisemaanlu** [comment]
There is definitely a memory leak, at least on low VRAM cards. I have a 1060 6GB and --medvram gets killed much more often than the regular mode with no cmd flags. Also I can't upscale res as much as I could on automatic1111, idk if its because pytorch is now 2.0 + cu118 or anything...

--medvram kills the process really fast, at least once in 10 mins. On base I can keep generating for what seems like an hour or so!

**Reply by vladmandic** [reply]
Id love to trace down memory leak - how much and during which operations exactly. You can monitor VRAM usage in real time in system info tab.

**Comment by usigg3** [comment]
There is a reference earlier in this thread regarding a discussion over in
the control net git. Seems there is some issue they are looking in to.


Den fre 28 apr. 2023 17:45Vladimir Mandic ***@***.***> skrev:

**Comment by BadWXfcst** [comment]
I disabled live preview and my VRAM leak appears to have disappear.

**Reply by vladmandic** [reply]
interesting. which live preview settings did you use?

**Reply by BadWXfcst** [reply]
Approx cheap, Combined

**Reply by vladmandic** [reply]
Can you try Approx NN? Its default and highly recommended to avoid race conditions leading to possible hangs (rare but possible) since tensor is accessed in multiple threads.

**Comment by mykeehu** [comment]
I just tested how much we consume from start to the end of the first image generation:
![image](https://user-images.githubusercontent.com/6390413/235310818-f8f637b7-8136-4258-9285-9e06a3658b99.png)

![image](https://user-images.githubusercontent.com/6390413/235310846-e3b632d3-9cf2-4f01-8989-8afc3c0cbf28.png)

The Hires Fix consumes a lot of video memory, but eventually shuts down. However, with Controlnet, I then loaded two model. One Canny and one Depth Zoe, and then see what happens:
![image](https://user-images.githubusercontent.com/6390413/235311266-b23895cb-7c6e-4c0f-8c4a-3f5b1097b4bb.png)

![image](https://user-images.githubusercontent.com/6390413/235311272-e554e4d9-4552-402f-aad5-9ba3ef9f2d3a.png)

It is clear that the Controlnet models does not clear itself from VRAM. The Model cache size is set to 1 in the settings, so it shouldn't stay there.
So there are two things that are hogging memory: the Hires Fix and Controlnet, which is not filling itself even though it is turned off.


And that's what happens when the Hires Fix dies:
![image](https://user-images.githubusercontent.com/6390413/235312046-bc03786a-abf4-4cff-ade1-8d3c052faccc.png)
![image](https://user-images.githubusercontent.com/6390413/235312038-fc3d904d-c2d2-4e1d-8947-c03d0eb5a243.png)

Is there any way to optimize the VRAM consumption of Hires Fix?

**Reply by GammelSami** [reply]
i think experienced the same yesterday and it was gone after reinstalling.

**Comment by vladmandic** [comment]
i've added built-in memory monitoring, run `webui --debug` and it will print stats upon completion of each operation.

**Reply by TKS4** [reply]
Hey vlad, I don't think this has anything to do with ControlNet, and has everything to do with this issue reported by this user on the main WebUI branch. https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/9922#issue-1688980247

I can also confirm I've had this issue for weeks even before making it to your awesome fork. But it has carried on over here. Though when I was experiencing the VRAM leak, I haven't the faintest of clues what the contributory factor was at all, but after trying to replicate it using this person's procedure, I've induced the issue without fail (or constant fail I should say) on a 1080ti

What happens most of the time when you finally fill up on VRAM (by getting one  of these sorts of failures) is the VRAM will not flush to the degree you can't even generate a 512x512 plain generation. Now, this isn't to say the other reports people are having about "previews slowly creeping up on VRAM" aren't true as well, and that ControlNet might have something to do with it (I doubt it because I've not been using ControlNet for days, and can still induce the VRAM leak with that person's instructions).

Restarting UI by terminating the webui.bat venv is the only way to flush whatever it is that's causing the VRAM to stay filled. Restarts through the UI do nothing. Oh and really quick since we're on the topic on UI restart buttons (this is tangential to another on-going issue). Is there any progress on the inability to use the "load from last prompt" button after a UI restart? You know, that whole issue where the UI becomes utterly unresponsive and also requires a full session restart?

Btw, I don't know where to say this but thank you for all your hard work, for this fork, and for your exceptional care and sustained effort. Please take it easy if you're able, most of us would be toast without you :]

**Comment by mykeehu** [comment]
I used the `--debug` option to do some generating with 2x Hires Fix. First without Controlnet, then with two Controlnet models (canny and depth zoe), then with one model (canny), finally switched off. So I should be in the same place as the first generation. 
However, I got a `retries: 7` value. Could this mean something?

```
LoRA weight_unet: 0.85, weight_tenc: 0.85, model: misstyyLora(006ebc678e12)
dimension: {128}, alpha: {128.0}, multiplier_unet: 0.85, multiplier_tenc: 0.85
create LoRA for Text Encoder: 72 modules.
create LoRA for U-Net: 192 modules.
original forward/weights is backed up.
enable LoRA for text encoder
enable LoRA for U-Net
shapes for 0 weights are converted.
LoRA model misstyyLora(006ebc678e12) loaded: <All keys matched successfully>
setting (or sd model) changed. new networks created.
100%|██████████████████████████████████████████| 50/50 [00:14<00:00,  3.56it/s]
100%|██████████████████████████████████████████| 30/30 [00:33<00:00,  1.12s/it]
22:17:17-997213 INFO     Processed: 1 Memory: {'ram': {'used': 10.89, 'total':
                         63.76}, 'gpu': {'used': 3.97, 'total': 12.0},
                         'retries': 0, 'oom': 0} txt
100%|██████████████████████████████████████████| 50/50 [00:10<00:00,  4.59it/s]
100%|██████████████████████████████████████████| 30/30 [00:33<00:00,  1.13s/it]
22:18:27-980456 INFO     Processed: 1 Memory: {'ram': {'used': 10.96, 'total':
                         63.76}, 'gpu': {'used': 3.99, 'total': 12.0},
                         'retries': 0, 'oom': 0} txt
100%|██████████████████████████████████████████| 50/50 [00:10<00:00,  4.65it/s]
100%|██████████████████████████████████████████| 30/30 [00:33<00:00,  1.12s/it]
22:19:41-743636 INFO     Processed: 1 Memory: {'ram': {'used': 10.93, 'total':
                         63.76}, 'gpu': {'used': 3.99, 'total': 12.0},
                         'retries': 0, 'oom': 0} txt
Loading model: control_v11p_sd15_canny [d14c016b]
Loaded state_dict from [I:\ControlNet models\control_v11p_sd15_canny.pth]
Loading config: I:\ControlNet models\control_v11p_sd15_canny.yaml
ControlNet model control_v11p_sd15_canny [d14c016b] loaded.
Loading preprocessor: canny
preprocessor resolution = 512
Loading model: control_v11f1p_sd15_depth [cfd03158]
Loaded state_dict from [I:\ControlNet models\control_v11f1p_sd15_depth.pth]
Loading config: I:\ControlNet models\control_v11f1p_sd15_depth.yaml
ControlNet model control_v11f1p_sd15_depth [cfd03158] loaded.
Loading preprocessor: depth_zoe
preprocessor resolution = 512
ControlNet preprocessor location: I:\annotator
img_size [384, 512]
Params passed to Resize transform:
        width:  512
        height:  384
        resize_target:  True
        keep_aspect_ratio:  True
        ensure_multiple_of:  32
        resize_method:  minimal
100%|██████████████████████████████████████████| 50/50 [00:16<00:00,  3.10it/s]
100%|██████████████████████████████████████████| 30/30 [00:53<00:00,  1.77s/it]
22:23:13-018557 INFO     Processed: 5 Memory: {'ram': {'used': 11.23, 'total':
                         63.76}, 'gpu': {'used': 7.07, 'total': 12.0},
                         'retries': 7, 'oom': 0} txt
Loading model: control_v11p_sd15_canny [d14c016b]
Loaded state_dict from [I:\ControlNet models\control_v11p_sd15_canny.pth]
Loading config: I:\ControlNet models\control_v11p_sd15_canny.yaml
ControlNet model control_v11p_sd15_canny [d14c016b] loaded.
Loading preprocessor: canny
preprocessor resolution = 512
100%|██████████████████████████████████████████| 50/50 [00:13<00:00,  3.74it/s]
100%|██████████████████████████████████████████| 30/30 [00:41<00:00,  1.40s/it]
22:25:04-867292 INFO     Processed: 3 Memory: {'ram': {'used': 12.54, 'total':
                         63.76}, 'gpu': {'used': 5.04, 'total': 12.0},
                         'retries': 7, 'oom': 0} txt
Loading model from cache: control_v11p_sd15_canny [d14c016b]
Loading preprocessor: canny
Pixel Perfect Mode Enabled.
resize_mode = ResizeMode.INNER_FIT
raw_H = 512
raw_W = 512
target_H = 512
target_W = 512
estimation = 512.0
preprocessor resolution = 512
100%|██████████████████████████████████████████| 50/50 [00:13<00:00,  3.67it/s]
100%|██████████████████████████████████████████| 30/30 [00:41<00:00,  1.39s/it]
22:27:42-620939 INFO     Processed: 3 Memory: {'ram': {'used': 12.52, 'total':
                         63.76}, 'gpu': {'used': 5.04, 'total': 12.0},
                         'retries': 7, 'oom': 0} txt
100%|██████████████████████████████████████████| 50/50 [00:10<00:00,  4.56it/s]
100%|██████████████████████████████████████████| 30/30 [00:34<00:00,  1.14s/it]
22:29:00-164651 INFO     Processed: 1 Memory: {'ram': {'used': 12.55, 'total':
                         63.76}, 'gpu': {'used': 5.04, 'total': 12.0},
                         'retries': 7, 'oom': 0} txt
```

I then switched the two Controlnets back on, and the retries were 14:
```
Loading model from cache: control_v11p_sd15_canny [d14c016b]
Loading preprocessor: canny
Pixel Perfect Mode Enabled.
resize_mode = ResizeMode.INNER_FIT
raw_H = 512
raw_W = 512
target_H = 512
target_W = 512
estimation = 512.0
preprocessor resolution = 512
Loading model: control_v11f1p_sd15_depth [cfd03158]
Loaded state_dict from [I:\ControlNet models\control_v11f1p_sd15_depth.pth]
Loading config: I:\ControlNet models\control_v11f1p_sd15_depth.yaml
ControlNet model control_v11f1p_sd15_depth [cfd03158] loaded.
Loading preprocessor: depth_zoe
Pixel Perfect Mode Enabled.
resize_mode = ResizeMode.INNER_FIT
raw_H = 512
raw_W = 512
target_H = 512
target_W = 512
estimation = 512.0
preprocessor resolution = 512
100%|██████████████████████████████████████████| 50/50 [00:16<00:00,  3.10it/s]
100%|██████████████████████████████████████████| 30/30 [00:52<00:00,  1.76s/it]
22:38:01-104750 INFO     Processed: 5 Memory: {'ram': {'used': 11.2, 'total':
                         63.76}, 'gpu': {'used': 7.31, 'total': 12.0},
                         'retries': 14, 'oom': 0} txt
```

**Reply by vladmandic** [reply]
Retries are how many times torch had to attempt memory allocation before it suceeded. If it didn't, oom counter would go up. Retries can be due to memory fragmentation.

But...there is too much going on here to make a clear conclusion.

I'd suggest clear restart, run 10 generations and shutdown server. Repeat for simple prompt, with Lora, with hires, with CN. Don't combine different things in single test until we get clear baseline.

**Reply by mykeehu** [reply]
I'll look at it in the next few days if someone doesn't do it first.

**Comment by mykeehu** [comment]
I've been looking at VRAM consumption over the last few days, and it seems that with the exception of the Controlnet models, both the Hires fix and the generating parts are filling up memory properly, so it's down to 3-3.5 GB for the first generation. For Controlnet, it eats 1-1.5 GB of VRAM per unit after background generation. The Hires Fix is interesting because it always pushes VRAM to the ceiling, but no more. If Controlnet is loaded, it will last up to 3 units, but because it gets less memory, it is a bit slower than without Controlnet.
So for now it works for me, and [I'm not sure if there will be a solution for Controlnet](https://github.com/Mikubill/sd-webui-controlnet/issues/990#issuecomment-1529313909), so I'll close this topic for my part.

**Comment by notfood** [comment]
I can't no longer reproduce after doing a pull to latest (today), just for information, last week I did a pull and Automatic was working okay, but after several generations it'd corrupt as if ControlNet was stuck no matter the setting. There was no error in console, just image/color corruption that would be fixed with a restart, then after 1 hour of working or so, the corruption would appear again. Restart, same gen, no corruption. No errors in console, the GPU appeared normal. Ram usage was normal.

This is no longer happening after today, just sharing the experience.
My settings: ArchLinux, automatic version `e0543e44` (the strange one), 22090d7b (the fixed one), Firefox 112.0.2
Images: TXT2IMG, ControlNet x3 in various inputs, usual text generation (UniPC, EulerA), no other extensions enabled other than the built-in.

**Comment by usigg3** [comment]
Can confirm it seems somehow to work now. I don't use high res fix but 2
control nets.

Den tors 4 maj 2023 23:03Meru ***@***.***> skrev:

**Reply by usigg3** [reply]
Seems some sort of memory error is back since the last days updates... It's worse than it was before. Memory is all used up after 10-20 generations with 2 controlnets.

**Reply by vladmandic** [reply]
how about without controlnet? if i'm about to troubleshoot, i need to reduce number of variables.

**Comment by usigg3** [comment]
I can try. If I monitor memory usage there is quite massive memory usage.
Before it used to be around 2 to 6 G but now it runs away to 8 to 9 pretty
quick before it crashes.

Den tors 11 maj 2023 19:17Vladimir Mandic ***@***.***> skrev:

**Reply by usigg3** [reply]
Did a restart of everything and windows decided it needed an update... Hasn't happened after that and python memory is in 2-3 GB again.

**Comment by gunnarsal** [comment]
Have experienced the same issue now with both A1111 and Vlad. GTX 1660 TI 6GB. It actually started ages ago. Was able to play around with multiple control nets and batch sizes, but at one point it is able to generate only 1 set, and then the memory usage gets so high that second one usually runs into CUDA error.

I reinstalled vlad just today and at first I was pleasantly surprised that after launch, it took atleast 1/4 if not 1/2 less then A1111 did, but right after the first Text2Image, it went up and after the second, the third ran out of memory. After I restart the server I can again run one generation :( 

I understand that most are using I2I where they are using the high res fixes and control net, but for me, it fills up with simple T2I as well. I Disabled Control net and am not using highres fix and the result is the same.

I hope a solution is found soon

**Comment by vladmandic** [comment]
@gunnarsal can you run webui with `--debug` flag and capture lines where it says `gc` (do not upload entire log) and describe what were you doing at the time of each line.

**Reply by gunnarsal** [reply]
> @gunnarsal can you run webui with `--debug` flag and capture lines where it says `gc` (do not upload entire log) and describe what were you doing at the time of each line.

Sorry, i do not quite know how to do it :) 

BTW, i seemed to have found a (temporary?) fix. When I moved the Face restore from VRAM to RAM, it does not fill up the VRAM after each generation and I am able to iterate more than one time.

**Reply by gunnarsal** [reply]
So some updates. I am able to use T2I without issues now, it frees up the vram nicelly. 

But unfortunately the program itself has gotten so "heavy" that it does not leave much room. This means that when I try to use I2I with even 1 controlNet and lowRam setting, it is running out of memory. It frees the memory so I do not need to restart, but unless control net gets some optimizations or some more stuff could be delegated to regular RAM or VRAM from 2 cards could be used For A1111 (as it is possible with oobabooga), then I will have half the features right now.
